library/src/main/java/com/android/volley/toolbox/DiskBasedCache.java

```diff
@@ -467,6 +467,11 @@ public static CacheHeader readHeader(InputStream is, boolean includeResponseHead
             entry.ttl = readLong(is);
             entry.softTtl = readLong(is);
 
+            // prune entries that are permacached
+            if (entry.ttl == Long.MAX_VALUE || entry.softTtl == Long.MAX_VALUE) {
+                throw new IOException();
```

Comment:
Is this a temporary thing just to get rid of old permacached things? If so leave a note/todo to remove this in the future and why we're doing it.